recipes/pathwatcher.go

```diff
@@ -0,0 +1,102 @@
+package recipes
+
+import (
+	"sync"
+
+	"github.com/betable/ezk"
+	"github.com/samuel/go-zookeeper/zk"
+)
+
+// PathWatcher is a recipe that watches for child events in a ZNode.
+// It will trigger an event when a child is created or deleted.
+//
+// Usage:
+// 	client := ezk.NewClient(ezk.ClientConfig{})
+// 	watcher := recipes.NewPathWatcher(client, "/")
+// 	watcher.Start()
+//	go func() {
+//		for {
+// 			select {
+// 				case e, ok := <-watcher.Event():
+// 				if !ok {
+// 					return
+// 				}
+// 				if e.Type == zk.EventNodeChildrenChanged {
+// 					// do something
+// 				}
+//
+// 				case err := <-monitorWatcher.Error():
+// 					// do something
+// 			}
+// 		}
+//	}()
+//	...
+//	// Stop the watcher
+//	watcher.Stop()
+type PathWatcher struct {
+	Path   string
+	client *ezk.Client
+	mut    sync.Mutex
+	event  chan zk.Event
+	stop   chan struct{}
+	done   chan struct{}
+	err    chan error
+}
+
+// NewPathWatcher creates a new PathWatcher on the passed path.
+func NewPathWatcher(client *ezk.Client, path string) *PathWatcher {
+	return &PathWatcher{
+		Path:   path,
+		client: client,
+		event:  make(chan zk.Event, 1),
```

Comment:
I'm a little worried about event being a buffered channel here. 

It's not that it is necessarily incorrect, but it is much trickier to reason about.

Why not have it be synchronous?  If we could get away with an unbuffered channel most of the time, that would be preferred.